src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java

```diff
@@ -185,29 +185,32 @@ public PostGobiOrdersHelper(HttpClientInterface httpClient,
       });
   }
 
-  public CompletableFuture<String> lookupDefaultLocationId(String defaultLocation) {
-    logger.info("No location received, using the default location");
-    //Always getting the first location until GOBI responds with how to handle locations for various orders
-      String query = HelperUtils.encodeValue(String.format(CQL_CODE_STRING_FMT, defaultLocation), logger);
-      String endpoint = String.format(LOCATIONS_ENDPOINT+QUERY, query);
-      return handleGetRequest(endpoint)
-        .thenApply(HelperUtils::extractLocationId)
-        .exceptionally(t -> {
-          logger.error("Exception looking up location id", t);
-          return null;
-        });
-
-  }
-
-  public CompletableFuture<List<String>> lookupMaterialTypeId(String materialType) {
-      String query = HelperUtils.encodeValue(String.format("name==%s", materialType), logger);
-      String endpoint = String.format(MATERIAL_TYPES_ENDPOINT+QUERY, query);
-      return handleGetRequest(endpoint)
-        .thenApply(HelperUtils::extractMaterialTypeId)
-        .exceptionally(t -> {
-          logger.error("Exception looking up material-type id", t);
-          return null;
-        });
+  /**
+   * Use the provided materialType. if one isn't provided, or the specified type
+   * can't be found, fallback to looking up "unspecified".If that fails,
+   * fallback to using the first one listed
+   *
+   * @param materialTypeCode
+   */
+  public CompletableFuture<List<String>> lookupMaterialTypeId(String materialTypeCode) {
```

Comment:
This is fine for now, I think we may want to pull fallback logic out of the lookup* methods and instead use nested mappings to define the fallback strategy.  The way it currently stands a tenant can't override this fallback strategy aside from (maybe?) being able to specify a different default...

Reply:
We can still use the nested mappings using the "defaultMappings" in the overrides , which would then be the fall back. 
Any example of how we want it to work?

Reply:
Let's just leave it as-is for now.  I think the point I was trying to make was that keeping the translations simple makes the system more flexible.  Incorporating more complex logic could result in boxing yourself in.